src/app/donation-start/donation-start-form/donation-tipping-slider/donation-tipping-slider.component.ts

```diff
@@ -178,18 +178,19 @@ export class DonationTippingSliderComponent implements OnInit, AfterContentInit,
     }
   }
 
-  updateHandlePosition(e: MouseEvent | TouchEvent | undefined) {
-    if(!e) { // the handle position is driven by the donation input change
-      this.calcAndSetPercentage();
-      this.pageX = 122;
-      this.mousePos = this.pageX - this.bar.nativeElement.offsetLeft - this.handle.nativeElement.offsetWidth / 2;
-      this.position = this.max * this.derivedPercentage / this.percentageEnd;
-      this.handle.nativeElement.style.marginLeft = this.position + 'px';
-    } else { // the handle position is driven by the mouse click on slider
-      this.mousePos = this.pageX - this.bar.nativeElement.offsetLeft - this.handle.nativeElement.offsetWidth / 2;
-      this.position = this.mousePos > this.max ? this.max : this.mousePos < 0 ? 0 : this.mousePos;
-      this.calcAndSetPercentage();
-    }
+  updateHandlePositionFromDonationInput() {
+    this.calcAndSetPercentage();
+    this.pageX = 122;
+    this.mousePos = this.pageX - this.bar.nativeElement.offsetLeft - this.handle.nativeElement.offsetWidth / 2;
+    this.position = this.max * this.derivedPercentage / this.percentageEnd;
+
+    this.handle.nativeElement.style.marginLeft = this.position + 'px';
+  }
+
+  updateHandlePositionFromClick() {
+    this.mousePos = this.pageX - this.bar.nativeElement.offsetLeft - this.handle.nativeElement.offsetWidth / 2;
```

Comment:
I think it would be worth getting rid of the pageX field and passing that info as a param instead. Page x is only accessed from these two functions and `move` which is the sole caller of `updateHandlePositionFromClick`